### ISSUE TITLE
torch_manual_seed does not seem to work with $sample()

### ISSUE THREAD

**Issue by LHBO** [mlverse/torch#937]
Hi,

This is my first time writing on GitHub, so I am unsure if I have included everything you need.

I have a question regarding reproducibility when generating random numbers from distributions.
From pages 36 and 37 in the manual (https://cran.r-project.org/web/packages/torch/torch.pdf) it seems that I should use the $sample() function, but then I am not able to set the seed and reproduce the sampled values even when setting the seed.
However,  setting the seed works when I use $rsample(). Is there a particular reason for this?

P.S.  I think you have done a splendid job in making torch much more available for us R users!

```R
library(torch)
dn = distr_normal(0, 1)

# These calls do NOT generate the same value
torch_manual_seed(10)
dn$sample(1)
torch_manual_seed(10)
dn$sample(1)

# These calls do NOT generate the same value (checking if I needed to set seed both in R and torch)
torch_manual_seed(10)
set.seed(10)
dn$sample(1)
torch_manual_seed(10)
set.seed(10)
dn$sample(1)

# These calls do generate the same value
torch_manual_seed(10)
dn$rsample(1)
torch_manual_seed(10)
dn$rsample(1)
```

**Comment by LHBO** [comment]
I should maybe add that I use torch version 0.9.0 (newest), R version 4.2.0 (2022-04-22), and I run on a mac with operating system x86_64-apple-darwin17.0.  

Manually setting the seed works for e.g. distr_categorical.

```R
library(torch)
dc = distr_categorical(torch_tensor(c(0.1, 0.5, 0.4)))

# Produce the same values
torch_manual_seed(1)
dc$sample(10)
torch_manual_seed(1)
dc$sample(10)
´´´

**Comment by dfalbel** [comment]
Hi @LHBO , thanks very much for reporting!

This is indeed a bug. It will be fixed once #938 is merged.
It will probably take some time to go into the CRAN version though.
Once the PR is merged you'll be able to install the dev version with 

```
remotes::install_github("mlverse/torch")
```
